mlflow/sklearn/__init__.py

```diff
@@ -756,14 +756,20 @@ def infer_model_signature(input_example):
         (X, y_true, sample_weight) = _get_args_for_metrics(estimator.fit, args, kwargs)
 
         # log common metrics and artifacts for estimators (classifier, regressor)
-        _log_estimator_content(
+        logged_metrics = _log_estimator_content(
             estimator=estimator,
             prefix=_TRAINING_PREFIX,
             run_id=mlflow.active_run().info.run_id,
             X=X,
             y_true=y_true,
             sample_weight=sample_weight,
         )
+        if y_true is None and not logged_metrics:
+            _logger.warning(
```

Comment:
In the future, we can likely add metrics for unsupervised techniques that don't depend on labels (e.g. https://scikit-learn.org/stable/modules/classes.html#pairwise-metrics for clustering). For now, printing a warning seems like a good idea, since users *can* pass labels to clustering methods and record metrics.